doc/optimizers.md

```diff
@@ -1563,6 +1563,71 @@ optimizer.Optimize(f, coordinates);
  * [SGD in Wikipedia](https://en.wikipedia.org/wiki/Stochastic_gradient_descent)
  * [Differentiable separable functions](#differentiable-separable-functions)
 
+## NSGA2
+
+*An optimizer for arbitrary multi-objective functions.*
+
+NSGA2 (Non-dominated Sorting Genetic Algorithm - II) is a multi-objective
+optimization algorithm. The algorithm works by generating a candidate population
+from a fixed starting point. At each stage of optimization, a new population of
+children is generated. This new population along with its predecessor is sorted
+using non-domination as the metric. Following this, the population is further
+segregated into fronts. A new population is generated from these fronts having
+size equal to that of the starting population.
+
+#### Constructors
+
+ * `NSGA2()`
+ * `NSGA2(`_`populationSize, maxGenerations, crossoverProb, mutationProb, mutationStrength, epsilon, lowerBound, upperBound`_`)`
+
+#### Attributes
+
+| **type** | **name** | **description** | **default** |
+|----------|----------|-----------------|-------------|
+| `size_t` | **`populationSize`** | The number of candidates in the population. This should be at least 4 in size and a multiple of 4. | `100` |
+| `size_t` | **`maxGenerations`** | The maximum number of generations allowed for NSGA2. | `2000` |
+| `double` | **`crossoverProb`** | Probability that a crossover will occur. | `0.6` |
+| `double` | **`mutationProb`** | Probability that a weight will get mutated. | `0.3` |
+| `double` | **`mutationStrength`** | The range of mutation noise to be added. This range is between 0 and mutationStrength. | `0.001` |
+| `double` | **`epsilon`** | The final value of the objective function for termination. If set to negative value, tolerance is not considered. | `1e-6` |
+| `arma::vec` | **`lowerBound`** | Lower bound of the coordinates of the initial population. | `1` |
+| `arma::vec` | **`upperBound`** | Lower bound of the coordinates of the initial population. | `1` |
```

Comment:
```suggestion
| `arma::vec` | **`upperBound`** | Upper bound of the coordinates of the initial population. | `1` |
```

Reply:
Also, are these coordinates of the initial population, or are they constraints on the coordinates of the whole population during the search process?  If it's the latter, then we should probably clarify in the documentation that `lowerBound` and `upperBound` really need to be set for the technique to operate as intended.

Reply:
Ah, right, so these are the constraints for the population for each of the variables involved during the search process. I think setting default parameters for these bounds do not make sense. 

I am temporarily setting `upperBound` to 1 and `lowerBound` to 0. 

I am also adding it in the documentation that these need to be explicitly initialized.